### DISCUSSION TITLE
Can I make DistroBox skip installing required packages?

### DISCUSSION THREAD

**Discussion by poperigby** [89luca89/distrobox#1113] [Q&A]
I have a custom Arch Linux image that I use, and it already has all the packages that DistroBox requires, so DistroBox trying to install them again is just lengthening in the creation time for no reason. Is there anyway to disable this?

**Comment by Fantome137** [comment]
I wish this had gotten answered. I just created an OCI container of Red Hat 5.2 ( form 1998 ). It does not have a package manager that can update over the Internet so distrobox fails trying to find one. I wish I knew how to tell it not to try.

**Comment by multiplealiases** [comment]
This is very hacky, but it's possible to replace the package manager with a blank file with the executable bit set using `--pre-init-hooks` in `distrobox create`. This is because Distrobox simply checks the return code of the package manager it selects, it does not do any further verification, and a blank shell script always returns 0.

For these examples, I'm using `apk` as the package manager.

If you already know all required packages are installed, set `--pre-init-hooks` to

```
mv /sbin/apk /sbin/apk.bak; touch /sbin/apk; chmod +x /sbin/apk
```

If, instead, you're trying to run an uncooperative container (e.g. everything but 1 package installs, but the last package kills the setup), you'll have to install the packages Distrobox needs (check the contents of `distrobox-init` for guidance) before pulling the rug on it. The following did the trick for me to start a Chimera Linux container that couldn't reinstall tzdata:

```
apk add bash bash bzip2 curl findutils gnupg iproute2 iputils keyutils less libcap ncurses pigz rsync shadow tcpdump tree unzip util-linux vulkan-loader xauth xz zip procps mount opendoas; mv /sbin/apk /sbin/apk.bak; touch /sbin/apk; chmod +x /sbin/apk
```

Then once Distrobox is done with setup, you can `mv /sbin/apk.bak /sbin/apk`.

If your container image doesn't have a package manager, just pretend that there is one. Set `--pre-init-hooks` to:

```
touch /sbin/apk; chmod +x /sbin/apk
```

**Reply by trevor-the-developer** [reply]
This is handy I had an issue where I had no internet access and I couldn't enter Distrobox
